release_dashboard/lib/widgets/progression.dart

```diff
@@ -13,48 +17,58 @@ import 'publish_tag_substeps.dart';
 import 'push_release_substep.dart';
 import 'release_completed.dart';
 
+/// First step of the release.
+///
+/// [ReleasePhase] from conductor core does not contain initialize release as a phase.
+/// But the release dashboard's first step is to initialize a release.
+/// This enum fills this missing step.
+// TODO(Yugue): [condcutor] Add InitializeRelease as the first phase to ReleasePhase,
+// https://github.com/flutter/flutter/issues/94765.
+enum ReleaseSteps {
+  initializeRelease,
+}
+
 /// Displays the progression and each step of the release from the conductor.
 ///
 // TODO(Yugue): Add documentation to explain
 // each step of the release, https://github.com/flutter/flutter/issues/90981.
 class MainProgression extends StatefulWidget {
   const MainProgression({
     Key? key,
-    this.previousCompletedStep,
   }) : super(key: key);
 
-  final int? previousCompletedStep;
-
   @override
   State<MainProgression> createState() => MainProgressionState();
 
-  static const List<String> _stepTitles = <String>[
-    'Initialize a New Flutter Release',
-    'Apply Engine Cherrypicks',
-    'Codesign Engine Binaries',
-    'Apply Framework Cherrypicks',
-    'Merge the Framework PR',
-    'Publish the Version Tag',
-    'Push and Verify the Release',
-    'Release is Completed',
-  ];
+  static const Map<Object, String> stepTitles = <Object, String>{
+    ReleaseSteps.initializeRelease: 'Initialize a New Flutter Release',
+    pb.ReleasePhase.APPLY_ENGINE_CHERRYPICKS: 'Apply Engine Cherrypicks',
+    pb.ReleasePhase.CODESIGN_ENGINE_BINARIES: 'Codesign Engine Binaries',
+    pb.ReleasePhase.APPLY_FRAMEWORK_CHERRYPICKS: 'Apply Framework Cherrypicks',
+    pb.ReleasePhase.PUBLISH_VERSION: 'Merge the Framework PR',
+    pb.ReleasePhase.PUBLISH_CHANNEL: 'Publish the Version Tag',
+    pb.ReleasePhase.VERIFY_RELEASE: 'Push and Verify the Release',
+    pb.ReleasePhase.RELEASE_COMPLETED: 'Release is Completed',
+  };
+
+  static const Map<Object, int> stepPosition = <Object, int>{
+    ReleaseSteps.initializeRelease: 0,
+    pb.ReleasePhase.APPLY_ENGINE_CHERRYPICKS: 1,
+    pb.ReleasePhase.CODESIGN_ENGINE_BINARIES: 2,
+    pb.ReleasePhase.APPLY_FRAMEWORK_CHERRYPICKS: 3,
+    pb.ReleasePhase.PUBLISH_VERSION: 4,
+    pb.ReleasePhase.PUBLISH_CHANNEL: 5,
+    pb.ReleasePhase.VERIFY_RELEASE: 6,
+    pb.ReleasePhase.RELEASE_COMPLETED: 7,
+  };
 }
 
 class MainProgressionState extends State<MainProgression> {
   int _completedStep = 0;
 
-  @override
-  void initState() {
-    // Enables the stepper to resume from the step it was left on previously.
-    if (widget.previousCompletedStep != null) {
```

Comment:
There is no need to have a parameter that receives `previousCompletedStep`, since the stepper tracks `currentPhase` of the state now.